### DISCUSSION TITLE
how to run MTM Kinematics Simulation

### DISCUSSION THREAD

**Discussion by JauJieYou** [jhu-dvrk/sawIntuitiveResearchKit#178] [Q&A] [answered]
The [Example of session](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/wiki/BuildROS2) is very clear and good. It could see clear of PSM1 movement. But there was only PSM1.rviz and PSM3.rviz in the subdirectory. How can I run the MTM simulation?  I know there are [other examples on ROS1](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/wiki/Kinematics-Simulation), Is there possible to convert it to ROS2?
  And for bilateral PSM simulation on the same screen, What should I do? 
  Is there anyone could give a demonstration or a class? 
Thanks a lot.

**Comment by adeguet1** [comment, marked as answer]
* For the MTMs, try `roslaunch dvrk_robot dvrk_arm_rviz.launch arm:=MTMR`.
* For ROS2, see https://github.com/jhu-dvrk/sawIntuitiveResearchKit/wiki/BuildROS2 - ROS2 is marginally supported at that point.
* I don't think one can use the `dvrk_arm_rviz` launch file to display two PSMs in Rviz.  Instead, you will need to create your own console JSON configuration file and use `rosrun dvrk_robot dvrk_console_json -j ...`.  See for example https://github.com/jhu-dvrk/sawIntuitiveResearchKit/blob/master/share/console/console-dVRK-cart-simulated.json.  Then you will have to load two URDF models in Rviz.  I've personally never done that but I'm sure it's not too hard and google can help you.

**Comment by JauJieYou** [comment]
Dear Professor:
  Thanks for your help. I am so appreciate and honored.
  I finally try as the ROSMA (
https://github.com/SurgicalRoboticsUMA/rosma_dataset) method on their web -
ROS part. It failed. Then I try the file in original dVRK git clone file
"""
roslaunch dvrk_robot dvrk_full_rviz.launch slave1:=PSM1 slave2:=PSM2
master1:=MTMR master2:=MTML
config:=/home/$(whoami)/catkin_ws/src/cisst-saw/sawIntuitiveResearchKit/share/console/console-full-system-simulated.json
"""
It still failed. It links PSM and MTM together.
[image: error4.png]
And the node is only one, instead of four (two PSM, two MTM)
[image: error4a.png]

I will keep try.
Thank for your help.

Best regards,
JJ You
